js/foam/browser/u2/StackView.js

```diff
@@ -0,0 +1,396 @@
+    },
+    {
+      name: 'views_',
+      documentation: 'Internal array of child views.',
+      factory: function() {
+        return [];
+      }
+    },
+    {
+      name: 'visibleStart_',
+      documentation: 'The index of the leftmost visible element.',
+      defaultValue: 0
+    },
+    {
+      name: 'visibleEnd_',
+      documentation: 'The index of the rightmost visible element.',
+      defaultValue: 0
+    },
+    {
+      name: 'noDecoration',
+      type: 'Boolean',
+      documentation: 'Set true to hide panel edges.',
+      defaultValue: false
+    },
+    {
+      model_: 'foam.core.types.StringEnumProperty',
+      name: 'transition',
+      documentation: 'Transition to use when adding or removing a view.',
+      defaultValue: 'slide',
+      choices: ['slide', 'fade']
+    },
+  ],
+
+  methods: [
+    function init() {
+      this.SUPER();
+      this.window.addEventListener('resize', this.onResize);
+    },
+    {
+      name: 'pushView',
+      documentation: 'Default pushView that works at the top level. See ' +
+          '$$DOC{ref:".pushView_"} for details.',
+      code: function(view, hints) {
+        this.pushView_(-1, view, hints);
+      }
+    },
+    {
+      name: 'popView',
+      documentation: 'Default popView that works at the top level. See ' +
+          '$$DOC{ref:".popView_"} for details.',
+      code: function() {
+        this.popView_(0);
+      }
+    },
+    {
+      name: 'popChildViews',
+      documentation: 'Default popChildViews that works at the top level. See ' +
+          '$$DOC{ref:".popChildViews_"} for details.',
+      code: function() {
+        this.popView_(1);
+      }
+    },
+    function elementAnimationAdd_(e) {
+      if (this.transition === 'slide') {
+        e.style({
+          width: '0px',
+          left: this.id$el.offsetWidth,
+          transition: 'left 300ms ease'
+        });
+      } else if (this.transition === 'fade') {
+        e.style({
+          opacity: 0,
+          transition: 'opacity 300ms ease'
+        });
+        // TODO(braden): Probably replace this with requestAnimationFrame?
+        this.setTimeout(function() { e.style({ opacity: 1 }); }, 50);
+      }
+    },
+    function elementAnimationRemove_(e) {
+      if (this.transition === 'slide') {
+        e.style({ left: this.id$el.offsetWidth });
+      } else if (this.transition === 'fade') {
+        e.style({ opacity: 0 });
+      }
+    },
+    function destroyChildViews_(index) {
+      while (this.views_.length > index) {
+        var obj = this.views_.pop();
+        this.publish(this.VIEW_DESTROYED, obj.content);
+        this.elementAnimationRemove_(obj.panel);
+        this.resize();
+        this.finishDestroy(obj.panel);
+      }
+    },
+    function finishDestroy(e) {
+      this.setTimeout(function() {
+        // Clean up after the animation.
+        // TODO(braden): Not sure if this is done?
+        //obj.view.remove();
+        e.unload();
+        e.destroy();
```

Comment:
I feel like we should feature request that Element.destroy() calls Element.unload if necessary.